modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/detail/ReferralDetail.js

```diff
@@ -59,43 +66,88 @@ export default function ReferralDetail({
 		return dateFormatters.formatChartTitle([firstDate, lastDate]);
 	}, [dateFormatters, firstDate, lastDate]);
 
-	const dispatch = useContext(ChartDispatchContext);
+	const dispatch = useContext(StoreDispatchContext);
+
+	const chartDispatch = useContext(ChartDispatchContext);
+
+	const {pieChartLoading, timeSpanKey, timeSpanOffset} = useContext(
+		ChartStateContext
+	);
 
-	const {timeSpanKey, timeSpanOffset} = useContext(ChartStateContext);
+	const {validAnalyticsConnection} = useContext(ConnectionContext);
 
 	const isPreviousPeriodButtonDisabled = useIsPreviousPeriodButtonDisabled();
 
-	const handleTimeSpanChange = (event) => {
-		const {value} = event.target;
+	const firstUpdate = useRef(true);
```

Comment:
I search for how to implement a useEffect that does not render in the first render and I found this: https://stackoverflow.com/questions/53253940/make-react-useeffect-hook-not-run-on-initial-render

We can implement a custom hook for this, but it will use a useRef too. As you said in another comment, maybe the problem of this is to receive in a prop all the data the first time the component renders. Maybe we should not receive this prop and execute the useEffect in the initial render too to obtain the data from the server, but that is something that will require back more back development.